test/others/criu-ns/run.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python
+
+import fcntl
+import os
+import pathlib
+import pty
+import shutil
+import subprocess
+import sys
+import termios
+import time
+
+
+CRIU_BIN = "../../../criu/criu"
+CRIU_NS = "../../../scripts/criu-ns"
+IMG_DIR = "dumpdir"
+DUMP_LOG = "dump.log"
+RESTORE_LOG = "restore.log"
+PIDFILE = "pidfile"
+
+
+def check_dumpdir(path=IMG_DIR):
+    if os.path.isdir(path):
+        shutil.rmtree(path)
+    os.mkdir(path, 0o755)
+
+
+def create_pty():
+    fd_m, fd_s = pty.openpty()
+    return (os.fdopen(fd_m, "wb"), os.fdopen(fd_s, "wb"))
+
+
+def create_isolated_dumpee():
+    pathlib.Path("running").touch()
+    fd_m, fd_s = create_pty()
+    pid = os.fork()
+    if pid == 0:
+        os.setsid()
+        os.dup2(fd_s.fileno(), 0)
+        os.dup2(fd_s.fileno(), 1)
+        os.dup2(fd_s.fileno(), 2)
+        fcntl.ioctl(fd_s.fileno(), termios.TIOCSCTTY, 1)
+        while True:
+            if not os.access("running", os.F_OK):
+                sys.exit(0)
+            time.sleep(1)
+    fd_m.close()
```

Comment:
Why do you have moved fd_m.close() here, why do you leave it open in child?

I would suggest closing everything which is possible in child, e.g after you setup 0,1,2 fds you can close both fd_m and fd_s, there is no point to have them after this point.

Reply:
thanks for the review @Snorch 

I had to move `fd_m.close()` when I made this terminal a controlling terminal by including `fcntl.ioctl(fd_s.fileno(), termios.TIOCSCTTY, 1)`. This is really weird I can't close `fd_m` in child with `fcntl.ioctl(fd_s.fileno(), termios.TIOCSCTTY, 1)` line in place. I'm using `Python 3.11.2` in my local machine.

Reply:
Can you close fd_m in child only, and leave it in parent? So that fd_m does is not saved into dump?

Reply:
> Can you close fd_m in child only, and leave it in parent?

No, with `fcntl.ioctl(fd_s.fileno(), termios.TIOCSCTTY, 1)` line in child, I'm unable to close `fd_m` in child. But, I can close or leave `fd_m` in parent.

Also, if I'm able to close `fd_m` in  child it's already closed in parent as well right.

> So that fd_m does is not saved into dump?

Sorry, I'm not sure what you mean :)

Reply:
Sorry, I see it's not possible to close it as I thought.

Reply:
Thanks @Snorch 
what should I do about [this](https://github.com/checkpoint-restore/criu/pull/2114#discussion_r1197288513)?